### PR TITLE
fix syslog pri parsing logic in syslog configs

### PR DESCRIPTION
**Pull request by badalkumar1223**
## Description
Updated syslog pri parsing logic to parse optional pri header.

## Related Issues
NA

## Todos
NA

### CODE REVIEW

File: config/enrichments/16_standardize_mac.conf [hunk 1]
```diff
@@ -65,6 +65,14 @@ filter {
         ]
       }
     }
+    if [related.mac] {
```
Comment (KrishnanandSingh) [inline config/enrichments/16_standardize_mac.conf:68]: this is added only in enrichment 17_related_mac so it doesn't exist here and is not required. Please remove

File: config/processors/syslog_list_inventory_tanium.conf [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright [2021] [Cargill, Incorporated.] 
+# Copyright [2021] [Cargill, Incorporated.]
```
Comment (KrishnanandSingh) [inline config/processors/syslog_list_inventory_tanium.conf:1]: this is an unrelated change. Please remove it from this PR.

**Review by KrishnanandSingh — CHANGES_REQUESTED** [review]

File: config/processors/syslog_list_inventory_tanium.conf [hunk 3]
```diff
@@ -13,89 +13,80 @@ filter {
...
-    }
-    json {
-     source => "rest_msg"
-     skip_on_invalid_json => true
-     target => "tanm"
-   }
+  # Parsing the syslog pri header if present
+  grok {
+    match => { "message" => "^.*?<(?<pri>\d+)>(\s)?(?<rest_msg>.*?)%" }
+  }
```
Comment (KrishnanandSingh) [inline config/processors/syslog_list_inventory_tanium.conf:19]: After the pri field is parsed with grok we need to do
```
  syslog_pri {
    syslog_pri_field_name => "pri"
    remove_field => [ "pri" ]
  }
  ```

File: config/processors/syslog_log_audit_a10.traffic.conf [hunk 4]
```diff
@@ -12,6 +12,10 @@ filter {
     add_field => { "observer.product" => "a10 loadbalancer" }
     add_field => { "observer.type" => "proxy" }
   }
+  # Parsing the syslog pri header if present
+  grok {
+    match => { "message" => "^.*?<(?<pri>\d+)>(\s)?(?<rest_msg>.*?)%" }
+  }
   mutate {
     strip => ["message"]
   }
```
Comment (KrishnanandSingh) [inline config/processors/syslog_log_audit_a10.traffic.conf:21]: update syslog_pri

File: config/processors/syslog_log_audit_checkpoint.fw.conf [hunk 5]
```diff
@@ -11,6 +11,10 @@ filter {
     # dropping timestamp from collector logstash so that it's regenrated in indexer
     remove_field => ["log_source_type", "@timestamp"]
   }
+# Parsing the syslog pri header if present
+  grok {
+    match => { "message" => "^.*?<(?<pri>\d+)>(\s)?(?<rest_msg>.*?)%" }
+  }
```
Comment (KrishnanandSingh) [inline config/processors/syslog_log_audit_checkpoint.fw.conf:18]: add syslog_pri part

**Review by KrishnanandSingh — CHANGES_REQUESTED** [review]
Badal can you please re-look at the changes. Here is what's required.
1. grok to get pri field
2. Use syslog_pri filter to parse pri
3. Make sure pri field is removed as it is not ecs.

Most of the files have pri field removed in the end. Please remove that. Use this instead
```
syslog_pri {
  syslog_pri_field_name => "pri" 
  remove_field => [ "pri" ]
}
```
It makes sense to drop extraneous fields just after we are done with them.
https://www.elastic.co/guide/en/logstash/7.2/plugins-filters-syslog_pri.html

File: config/processors/syslog_log_audit_checkpoint.fw.conf (hunk 5 above)
Comment (badalkumar1223) [reply config/processors/syslog_log_audit_checkpoint.fw.conf:22]: added syslog_pri field in .conf files, please check.

**Comment by nnovaes** [comment]
> Badal can you please re-look at the changes. Here is what's required.
> 
> 1. grok to get pri field
> 2. Use syslog_pri filter to parse pri
> 3. Make sure pri field is removed as it is not ecs.
> 
> Most of the files have pri field removed in the end. Please remove that. Use this instead
> 
> ```
> syslog_pri {
>   syslog_pri_field_name => "pri" 
>   remove_field => [ "pri" ]
> }
> ```
> 
> It makes sense to drop extraneous fields just after we are done with them.
> https://www.elastic.co/guide/en/logstash/7.2/plugins-filters-syslog_pri.html

not must but nice to have:

1. we could have the grok parse the field to **syslog_pri** , i.e.

```
 grok {
    match => { "message" => "^.*?<(?<syslog_pri>\d+)>(\s)?(?<rest_msg>.*?)%" }
  }

```
and then just do the filter as 

`syslog_pri {}`

2. also, the grok field you have assumes we'll get a valid syslog_pri and is a general good assumption, but maybe replacing \d+ with \d{1,3} is more accurate. syslog pri ranges between 0 to 191, but maybe doing the regex to catch 0 to 191 only is going to be too much of a complication.

**Comment by KrishnanandSingh** [comment]
@nnovaes can you fix the conflixt for config/processors/syslog_log_audit_cisco.dna.conf?

**Review by brian-grabau — DISMISSED** [review]

**Comment by nnovaes** [comment]
> @nnovaes Can you please fix the conflicts for cisco configs? For the guardduty config, we need both changes.

@KrishnanandSingh. This is @badalkumar1223 fork. He needs to sync it.

**Review by Akhila-Y — CHANGES_REQUESTED** [review]

File: config/processors/syslog_log_security_palo.alto.fw.conf [hunk 6]
```diff
@@ -44,7 +33,7 @@ filter {
 
     dissect {
       mapping => {
-        "message" => "%{log_time} {%{?data}} <%{?pri}>%{?data} %{?data} %{?data} %{host.hostname} %{?data}|%{event.provider}|%{?data}|%{rule.version}|%{rule.category}|%{event.type}|%{event.severity}|%{somedata}"
+        "actual_msg" => "%{host.hostname} %{?data}|%{event.provider}|%{?data}|%{rule.version}|%{rule.category}|%{event.type}|%{event.severity}|%{somedata}"
```
Comment (Akhila-Y) [inline config/processors/syslog_log_security_palo.alto.fw.conf:36]: pattern mismatches with the "actual_msg" format

File: config/processors/syslog_log_security_sdwan.app.conf [hunk 7]
```diff
@@ -6,45 +6,34 @@ input {
...
-  }
+    
   mutate{
     remove_field => ["pri"]
   }
     # 1. Start by copying the original message to log.original.  We'll then pick the common fields and placing them under [tmp]. [tmp] is a temporary slot to work out the parsing
 
     mutate {
-        add_field => {"[log.original]" => "%{[message]}"}
+        add_field => {"[log.original]" => "%{[actual_msg]}"}
```
Comment (Akhila-Y) [inline config/processors/syslog_log_security_sdwan.app.conf:31]: "log.original" is always a copy of "message" but not "actual_msg"

File: config/processors/syslog_log_security_symantec.dlp.conf [hunk 8]
```diff
@@ -9,9 +9,18 @@ filter {
...
+    match => { "message" => "^.*?<(?<pri>\d+)>(\s)?(?<actual_msg>.*)" }
+  }
+  syslog_pri {
+  syslog_pri_field_name => "pri" 
+  remove_field => [ "pri" ]
+  }
   dissect {
     mapping => {
-      "message" => "%{?log.date_time} {%{?data}} <%{?pri}>%{event.created->} %{+event.created} %{+event.created} %{observer.hostname} %{?data}|%{?data}|%{?data}|%{?data}|%{rule.ruleset}|%{rule.name}|%{?data}|%{somedata}"
+      "actual_msg" => "%{?log.date_time} {%{?data}} <%{?pri}>%{event.created->} %{+event.created} %{+event.created} %{observer.hostname} %{?data}|%{?data}|%{?data}|%{?data}|%{rule.ruleset}|%{rule.name}|%{?data}|%{somedata}"
```
Comment (Akhila-Y) [inline config/processors/syslog_log_security_symantec.dlp.conf:23]: change dissect pattern to "actual_msg" format.

**Review by Akhila-Y — CHANGES_REQUESTED** [review]

File: config/processors/syslog_log_security_layer7.securespan.soa.gw.conf [hunk 9]
```diff
@@ -5,7 +5,16 @@ input {
...
 }
+
 filter {
+  # Parsing the syslog pri header if present
+  grok {
+    match => { "message" => "^.*?<(?<pri>\d+)>(\s)?(?<actual_msg>.*)" }
+  }
+  syslog_pri {
+  syslog_pri_field_name => "pri" 
+  remove_field => [ "pri" ]
```
Comment (Akhila-Y) [inline config/processors/syslog_log_security_layer7.securespan.soa.gw.conf:16]: we are trying to compare pri in the later code, so dont remove it here, pri can be removed at end

**Review by Akhila-Y — CHANGES_REQUESTED** [review]

**Comment by KrishnanandSingh** [comment]
Closing it as it was redone in #115
